src/main/kotlin/org/jetbrains/teamcity/github/controllers/GitHubWebHookListener.kt

```diff
@@ -284,24 +261,24 @@ class GitHubWebHookListener(private val WebControllerManager: WebControllerManag
                 PullRequestMergeBranchChecker.schedule(info, hookInfo, user, id)
             }
         }
-        return doForwardToRestApi(info, user, request, response)
+        return scheduleChangesCollection(repository, user)
     }
 
-    private fun doForwardToRestApi(info: GitHubRepositoryInfo, user: UserEx, request: HttpServletRequest, response: HttpServletResponse): Pair<Int, String>? {
-        val httpId = info.id
-        val sshId = info.server + ":" + info.owner + "/" + info.name
-        val dispatcher = request.getRequestDispatcher("/app/rest/vcs-root-instances/commitHookNotification?" +
-                                                      "locator=vcsRoot:(type:jetbrains.git,count:99999),or:(property:(name:url,value:$httpId,matchType:contains,ignoreCase:true),property:(name:url,value:$sshId,matchType:contains,ignoreCase:true)),count:99999")
-        if (dispatcher != null) {
-            val layered = LayeredHttpServletRequest(request)
-            SessionUser.setUser(layered, user)
-            layered.setAttribute("INTERNAL_REQUEST", true)
-            SecurityContext.runAs(user) {
-                dispatcher.forward(layered, response)
+    private fun scheduleChangesCollection(repository: Repository, user: UserEx): Pair<Int, String>? {
+        return SecurityContext.runAs<Pair<Int, String>>(user) {
+            val vcsRoots = webhooksHelper.findRelevantVcsRootInstances(repository)
+            if (vcsRoots.isEmpty())
+                SC_OK to "No relevant VCS roots found"
+            else {
+                webhooksHelper.checkForChanges(vcsRoots)
+                val vcsRootIdSet = vcsRoots.map { it.parent.externalId }.toSet()
+                val vcsRootIds = vcsRootIdSet.joinToString("\n")
+
+               SC_OK to "Checking for changes scheduled for ${vcsRoots.size} " +
```

Comment:
If I am not mistaken in case of REST it returns 202, but not sure it's that important here.

Reply:
I will check, but 202 makes sense indeed

Reply:
Changed to 202, thanks